src/Microsoft.Extensions.Configuration.Json/JsonConfigurationProvider.cs

```diff
@@ -52,18 +52,20 @@ public override void Load(Stream stream)
 
         private static string RetrieveErrorContext(JsonReaderException e, IEnumerable<string> fileContent)
         {
-            string errorLine;
+            string errorLine = null;
             if (e.LineNumber >= 2)
             {
                 var errorContext = fileContent.Skip(e.LineNumber - 2).Take(2).ToList();
-                errorLine = errorContext[0].Trim() + Environment.NewLine + errorContext[1].Trim();
+                if (errorContext.Count() >= 2)
```

Comment:
Anyway, I don't feel strongly that we need to do much more than what you did to prevent the NRE. As you said this is a best effort to print some context on where in the file the error is, but if the exception explains that the end if file was unexpectedly encountered then that context may be unnecessary.